src/plugins/telemetryTable/components/table.vue

```diff
@@ -894,6 +938,45 @@ export default {
             this.isAutosizeEnabled = true;
 
             this.$nextTick().then(this.calculateColumnWidths);
+        },
+        getViewKey() {
+            return this.viewKey;
+        },
+        getViewContext() {
+            return {
+                getViewKey: this.getViewKey,
+                exportAllDataAsCSV: this.exportAllDataAsCSV,
+                exportMarkedRows: this.exportMarkedRows,
+                unmarkAllRows: this.unmarkAllRows,
+                togglePauseByButton: this.togglePauseByButton,
+                expandColumns: this.recalculateColumnWidths,
+                autosizeColumns: this.autosizeColumns
+            };
+        },
+        initializeViewActions() {
+            if (this.markedRows.length > 0) {
```

Comment:
I'm just wondering if there's some way that actions could provide defaults themselves, rather than the view having to configure hidden/shown and enabled/disabled state on load. Requiring the view to do this on load makes loading the initial action menu state asynchronous, which complicates things and necessitates debouncing etc. 

eg. it would be great if views could just call something like `ActionCollection.getVisible()` or `ActionCollection.getAll()` to get a reasonable initial state synchronously. 

If it depends upon some persisted state of the object, we could define an optional `init(objectPath)` function on actions where the object is passed into it and the Action can initialize its own visible / hidden state.

Reply:
There may be a small issue in doing it that way, since actionCollection is shared by the view and the browseBar. The browseBar will collect all the actions before the view has been mounted/initialized, then the view can use the same actionCollection to disable/hide/show any actions it sees fit according to its initialized data. One way would be to have an initialize function on all these view actions, that would use the viewContext to set properties. But that seems to complicate the action registering process